### PR TITLE
Add /devfile PUT and GET endpoints

### LINKED ISSUE #6901: Add API endpoints to get and save the local Devfile
/kind task
/area dev

API endpoints to get and save the local Devfile (to initialize the `odo dev` web UI state)

## Acceptance Criteria 
- [ ] It should expose a `GET` endpoint to fetch the local Devfile from the current folder where `odo dev` is running from
- [ ] It should expose a `PUT` endpoint that would save a given Devfile and persist it into the local folder where `odo dev` is running from
- [ ] It should work with the raw Devfile
- [ ] It should validate the Devfile before writing it
- [ ] UI should use these endpoints to load and save the Devfile

## Links 
- Related Epic (mandatory): https://github.com/redhat-developer/odo/issues/6895

## Requires
- https://github.com/redhat-developer/odo/issues/6301

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this:**

/kind feature

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #6901

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: ui/src/app/api-gen/.openapi-generator/FILES
Comment (rm3l) [inline ui/src/app/api-gen/.openapi-generator/FILES:1]: One side note: can you add `ui/src/app/api-gen` to the [`.gitattributes`](https://github.com/redhat-developer/odo/blob/main/.gitattributes) file, to hide by default in the PR changes view?

File: pkg/apiserver-impl/api_default_service.go [hunk 1]
```diff
@@ -95,3 +103,53 @@ func (s *DefaultApiService) InstanceGet(ctx context.Context) (openapi.ImplRespon
...
+	return openapi.Response(http.StatusOK, openapi.DevfileGet200Response{
+		Content: string(content),
+	}), nil
+
+}
+
+func (s *DefaultApiService) DevfilePut(ctx context.Context, params openapi.DevfilePutRequest) (openapi.ImplResponse, error) {
+	devfilePath := odocontext.GetDevfilePath(ctx)
+
+	err := s.validateDevfile(ctx)
```
[the remark this answers is not in the document]
Comment (feloy) [reply pkg/apiserver-impl/api_default_service.go]: Oups, my bad :)

**Review by rm3l — APPROVED** [review]
Works great - thanks!

**Comment by feloy** [comment]
/override windows-integration-test/Windows-test
```
 [FAILED] [2.625 seconds]
odo create/delete/list/set namespace/project tests delete namespace when force-deleting a valid namespace [BeforeEach] should successfully delete the namespace asynchronously
  [BeforeEach] C:/Users/Administrator.ANSIBLE-TEST-VS/4341/tests/integration/cmd_namespace_test.go:74
  [It] C:/Users/Administrator.ANSIBLE-TEST-VS/4341/tests/integration/cmd_namespace_test.go:96
```

**Pull request merged**